grassroots-frontend/src/routes/SharedSearch.tsx

```diff
@@ -0,0 +1,37 @@
+import { createFileRoute } from "@tanstack/react-router";
+import { JSX, useState } from "react";
+import { useContactSearch } from "../hooks/useContactSearch";
+import { PaginatedContacts } from "../components/PaginatedContacts";
+import { castWithConversion } from "../grassroots-shared/Cast";
+import { ContactSearchInDTO } from "../grassroots-shared/Contact.entity.dto";
+
+export const Route = createFileRoute("/SharedSearch")({
+  component: SharedSearch,
+  validateSearch: (search: Record<string, unknown>): ContactSearchInDTO => {
+    search.id = search.id == "" ? undefined : search.id;
+    return castWithConversion(ContactSearchInDTO, search);
+  },
+});
+
+const ROWS_PER_PAGE = 10;
+
+function SharedSearch(): JSX.Element | null {
+  const search = Route.useSearch();
+  const [rowsToSkip, setRowsToSkip] = useState<number>(0);
+
+  const { data: results } = useContactSearch({
+    contact: search,
+    paginated: {
+      rowsToSkip,
+      rowsToTake: ROWS_PER_PAGE,
+    },
+  });
+  return results ? (
+    <PaginatedContacts
+      contacts={results.contacts}
+      paginated={results.paginated}
+      setRowsToSkip={setRowsToSkip}
+      rowsPerPage={ROWS_PER_PAGE}
+    ></PaginatedContacts>
+  ) : null;
```

Comment:
Not required for this to merge, but should we have a loading component and a no results component?

Reply:
Just to clarify before I write this - I agree we should display something reasonable in the "loading" and "no results" cases. I typically wouldn't bother to pull them out into separate components from the PaginatedContacts component. Do you think separate components makes sense?

Reply:
I don't have an opinion on component vs integrated at this time